### DISCUSSION TITLE
Semantics for (Step-indexed) Kripke Models with assumptions on lhs

### DISCUSSION THREAD

**Discussion by Florian3k** [ppolesiuk/type-systems-notes#94] [Q&A] [answered]
During last lecture we didn't define what $\Gamma\models\phi$ means, only $\models\phi$.
We need this (I think) for exercise 4 and 6.

Can we treat $\triangleright\Sigma,\Gamma\models\phi$ as $\models\triangleright\Sigma'\to\Gamma'\to\phi$ (where $X'$ means all assumptions from $X$ joined using $\land$)? Those should be equivalent via implication introduction/elimination and import/export law.

Can we prove eg. $\models\triangleright\phi\to\psi\to\phi \implies \models\psi\to\phi$ instead of $\Gamma,\triangleright\phi\models\phi \implies \Gamma\models\phi$?

cc @ppolesiuk

**Comment by ppolesiuk** [comment, marked as answer]
It's almost the same. We define $\Gamma \models \varphi$ to hold iff for each valuation $\sigma$ and world $w$ such that $\sigma,w\models \Gamma$ we have $\sigma,w \models \varphi$.
